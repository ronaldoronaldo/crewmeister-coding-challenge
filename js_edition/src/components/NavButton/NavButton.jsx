import React from "react";
import {NavButtonStyle, ButtonTextStyle} from "./NavButton.style";

export default function NavButton ({ onClick, text, Icon }) {
  return (
    <NavButtonStyle
      onClick={onClick}>
      <ButtonTextStyle>
        {text}
      </ButtonTextStyle>
      <Icon/>
    </NavButtonStyle>
  )
}

