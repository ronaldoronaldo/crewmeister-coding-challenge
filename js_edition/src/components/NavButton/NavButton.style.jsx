import styled from 'styled-components'
import {spacing, screens} from "../../styles/styleGuide";
import {ButtonStyle, ButtonText} from "../../styles/commun";

export const ButtonTextStyle = styled(ButtonText)`
  display: none;
  @media (min-width: ${screens.medium}) {
    display: flex;
  }
`

export const NavButtonStyle = styled(ButtonStyle)`
  margin-right: ${spacing.large};
  @media (min-width: ${screens.medium}) {
    margin-right: ${spacing.xLarge};
  }

`
