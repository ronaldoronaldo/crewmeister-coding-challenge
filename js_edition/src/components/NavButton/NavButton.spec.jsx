import React from "react";
import { render } from "@testing-library/react";
import GetAppIcon from '@material-ui/icons/GetApp';

import NavButton from "./NavButton";

describe("<NavButton />", () => {
  it("should match snapshot", () => {
    const dummyFunction = () => {}
    const { container } = render(<NavButton text="myBtn" onClick={dummyFunction} Icon={GetAppIcon}/>);
    expect(container).toMatchSnapshot();
  });
});
