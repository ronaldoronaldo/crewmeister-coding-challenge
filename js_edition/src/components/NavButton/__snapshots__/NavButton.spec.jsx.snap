// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButton /> should match snapshot 1`] = `
<div>
  <button
    class="sc-AxirZ sc-AxhCb kYuvxj"
  >
    <span
      class="sc-AxjAm sc-AxiKw QVUtw"
    >
      myBtn
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
      />
    </svg>
  </button>
</div>
`;
